run/radius2john.py

```diff
@@ -1,68 +1,163 @@
+    )
 
-from binascii import hexlify
-from sys import argv
-import os
+    dumped_ips[ip] = "request"
 
-if len(argv) < 2:
-    print('Usage: ./radius2john.py <pcap files>')
-    exit(1)
 
-filenames = argv[1:]
+if __name__ == "__main__":
 
-for filename in filenames:
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter,epilog=
+    """
+    ### Utility to bruteforce RADIUS shared-secret written by k4amos
+    Basic Usage: ./radius2john.py <pcap files>"
 
-    capture_file = rdpcap(filename)
+    ---
 
-    accounting_request_packets = {}
-    accounting_response_packets = {}
+    Application of two  methods described in http://www.untruth.org/~josh/security/radius/radius-auth.html :
+    - "3.3 User-Password Attribute Based Shared Secret Attack"
+    - "3.1 Response Authenticator Based Shared Secret Attack"
 
-    for packet in capture_file:
+    # For attack 3.3 :
+    We try authentications using a known password, and sniff the radius packets to a pcap file.
+    This script reads access-request in the pcap file, and dumps the md5(RA+secret) and RA, in a john-friendly format.
 
-        if Radius in packet:
-            src_ip = packet[IP].src
-            dst_ip = packet[IP].dst
+    # For attack 3.1:
+    We don't need to try authentications. Just sniff the radius packets in a pcap file.
+    This script reads the pcap file, matches radius responses with the corresponding all_requests,
+    and dumps md5 and salt as needed.
+    		""")
 
-            if packet[Radius].code == 4 and not (src_ip, dst_ip) in accounting_request_packets:
-                accounting_request_packets[(src_ip, dst_ip)] = packet
-            elif packet[Radius].code == 5 and not (dst_ip, src_ip) in accounting_response_packets:
-                accounting_response_packets[(dst_ip, src_ip)] = packet
+    parser.add_argument('-f', '--file', type=str, required=True, nargs='+')
+    parser.add_argument('--single', help='To get only one hash per client IPs', action='store_true', default=False)
+    parser.add_argument('-l', '--login', type=str,help='User login used for the 3.3 attack')
+    parser.add_argument('-p', '--password', type=str, help='User password used for the 3.3 attack')
+    
 
-    for ips, packet in accounting_response_packets.items():
-        if ips in accounting_request_packets:
-            code = hex(packet[Radius].code)[2:].zfill(2)
-            identifier = hex(packet[Radius].id)[2:].zfill(2)
-            length = hex(packet[Radius].len)[2:].zfill(4)
-            authenticator = hexlify(accounting_request_packets[ips][Radius].authenticator).decode('utf-8')
 
-            salt = code + identifier + length + authenticator
+    parsed_args = parser.parse_args()
+    args = vars(parsed_args)
 
-            hash_content = hexlify(packet[Radius].authenticator).decode('utf-8')
-            hash_type = 1009 if len(salt) <= 16 else 1017
+    if args["login"] is not None and args["password"] is None:
```

Comment:
Did I understand correctly, that you always need both, login and password for attack 3.3? If yes, then also a check for password is not None but login is would be required, wouldn't it?